rectools/models/base.py

```diff
@@ -398,7 +398,9 @@ def recommend(
         """
         self._check_is_fitted()
         self._check_k(k)
-
+        # `dataset.user_id_map.external_dtype`, `dataset.item_id_map.external_dtype` can change
```

Comment:
Why can they change?

It's not allowed to change function param

Reply:
We are not changing dataset itself. But we do call `dataset = self._custom_transform_dataset_u2i(dataset, users, on_unsupported_targets)`
and here we get new object under `dataset` name.

Reply:
fixed